cpp/log/tree_signer-inl.h

```diff
@@ -189,10 +189,13 @@ util::Status TreeSigner<Logged>::SequenceNewEntries() {
   }
 
   // Sanity check: make sure no hashes above the serving_sth level vanished:
-  const uint64_t serving_tree_size(serving_sth.ValueOrDie().tree_size());
+  CHECK_LE(serving_sth.ValueOrDie().tree_size(), INT64_MAX);
+  const int64_t serving_tree_size(serving_sth.ValueOrDie().tree_size());
   for (const auto& s : sequenced_hashes) {
     if (!s.second.second /*present*/) {
       // if it disappeared, check it's underwater:
+      // This is safe because the tree cannot get large enough to overflow a
```

Comment:
Probably don't need this comment now as we assert that it's not going to overflow above.